### PR TITLE
[REF] Reorganize into blueprints + add some sphinx docs

### PR DESCRIPTION
**Pull request by DESm1th**
I still need to add a bunch of documentation and move the readme contents into the sphinx docs but it looks like I wont get to come back to this for a week or more so I'll add it later, since this has been sitting idle for about a week already. 

The 'undefined name' errors from queries.py are all 'metric' related stuff that needs to be rehabilitated by us later. I'm leaving all of that alone for now instead of just ripping it out.

Also, note that most of the changes are just from reorganizing existing code. The only new pieces are:

1. The contents of the docs folder (much of which is autogenerated by sphinx) and doc strings
2. ContextThreadExecutor in task_scheduler.py
3. The contents of dashboard/__init__.py
4. Changes to the logging configuration in config/logging.py

### CODE REVIEW

**Comment by DESm1th** [comment]
The pep8speaks issues are all things that should have been ignored based on our .flake8 settings file. I suspect that it's not correctly parsing the settings because we also have a .pep8speaks.yml settings file (apparently it SHOULD read it, but it appears not to be doing so). Either way, those errors are all false alarms and should be ignored.

Also, the two remaining codacy 'issues' are not real issues ('copyright' is a sphinx doc setting and can't be changed, and the listening on all interfaces thing is intentional/not used in production/only run manually by users).

File: dashboard/datman_utils.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    if not scan.bids_name:
+        return
+
+    bids_file = datman.scanid.parse_bids_filename(scan.bids_name)
+    sub_dirs = []
+    sub_dirs.append(subject_folder)
+    sub_dirs.append(session_folder)
+    for path, dirs, files in os.walk(session_folder):
+        for dir in dirs:
```
Comment (josephmje) [inline dashboard/datman_utils.py]: use another variable name instead of dir

File: dashboard/datman_utils.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+            sub_dirs.append(os.path.join(path, dir))
+
+        for item in files:
+            if bids_file == item:
+                full_path = os.path.join(path, item)
+                os.remove(full_path)
+
+    # Clean up any folders that may now be empty
+    for dir in reversed(sub_dirs):
```
Comment (josephmje) [inline dashboard/datman_utils.py]: use another variable name instead of dir

File: dashboard/datman_utils.py (hunk 1 above)
Comment (DESm1th) [reply dashboard/datman_utils.py]: Ooh nice catch! I missed that (and so did codacy)

**Comment by DESm1th** [comment]
@jerdra @josephmje Hey guys! Any last feedback before merging this? Are you guys ok with me moving ahead?

**Comment by josephmje** [comment]
> @jerdra @josephmje Hey guys! Any last feedback before merging this? Are you guys ok with me moving ahead?

No additional feedback from me.

**Comment by jerdra** [comment]
> @jerdra @josephmje Hey guys! Any last feedback before merging this? Are you guys ok with me moving ahead?

will look through today! sorr working through my PR backlog :see_no_evil:

File: README.md [hunk 3]
```diff
@@ -36,25 +36,30 @@ QC metrics generated by our nightly pipelines.
...
 
 ## Dashboard Structure
-[dashboard/models.py](dashboard/models.py) defines the _Object Relational Mapping (ORM)_ used by
+[dashboard/models](dashboard/models) defines the _Object Relational Mapping (ORM)_ used by
 SQLAlchemy to map from the relational database to the python objects used in
 the code.
 
-[dashboard/views.py](dashboard/views.py) defines the _entry points_ (i.e. valid URLs for the app).
+All of the 'views.py' scripts define the _entry points_ (i.e. valid URLs for
+the app). These have been organized into blueprints that group related
```
Comment (jerdra) [inline README.md:56]:
```suggestion
the app). These have been organized into Flask blueprints that group related
```

**Pull request merged**